cli/src/pcluster/lib/lib.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not
+# use this file except in compliance with the License. A copy of the License is
+# located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or
+# implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import yaml
+
+import pcluster.api.controllers.cluster_operations_controller
+import pcluster.api.errors
+import pcluster.cli.model
+from pcluster.cli.entrypoint import dispatch, param_coerce
+from pcluster.utils import to_snake_case
+
+
+def _gen_class(model):
```

Comment:
[optional] adding typing info to this module would improve navigating through the generation functions.

Reply:
I added some typing -- the real help is explained in the other file that generates the model so I added a comment pointing to that (as well).